src/coprocessor/local_metrics.rs

```diff
@@ -54,3 +80,203 @@ impl ScanCounter {
+        statics.add_statistics(other);
+    }
+
+    pub fn flush(&mut self) {
+        for (type_str, v) in self.data.drain() {
+            for (cf, details) in v.stat.details() {
+                for (tag, count) in details {
+                    COPR_SCAN_DETAILS
+                        .with_label_values(&[type_str, cf, tag])
+                        .inc_by(count as f64)
+                        .unwrap();
+                }
+            }
+        }
+    }
+}
+
+/// `CopFlowStatistics` is for flow statistics, it would been reported to Pd by flush.
+pub struct CopFlowStatistics {
+    data: HashMap<u64, FlowStatistics>,
+    sender: FutureScheduler<PdTask>,
+}
+
+impl CopFlowStatistics {
+    pub fn new(sender: FutureScheduler<PdTask>) -> CopFlowStatistics {
+        CopFlowStatistics {
+            sender: sender,
+            data: Default::default(),
+        }
+    }
+
+    pub fn add(&mut self, region_id: u64, stats: &Statistics) {
+        let flow_stats = self.data
+            .entry(region_id)
+            .or_insert_with(FlowStatistics::default);
+        flow_stats.add(&stats.write.flow_stats);
+        flow_stats.add(&stats.data.flow_stats);
+    }
+
+    pub fn flush(&mut self) {
+        if self.data.is_empty() {
+            return;
+        }
+        let mut to_send_stats = HashMap::default();
+        mem::swap(&mut to_send_stats, &mut self.data);
+        if let Err(e) = self.sender.schedule(PdTask::ReadStats {
+            read_stats: to_send_stats,
+        }) {
+            error!("send coprocessor statistics: {:?}", e);
+        };
+    }
+}
+
+/// `ExecLocalMetrics` collects metrics from `ExectorMetrics`
+pub struct ExecLocalMetrics {
+    flow_stats: CopFlowStatistics,
+    scan_details: ScanDetails,
+    scan_counter: ScanCounter,
+    exec_counter: ExecCounter,
+}
+
+impl ExecLocalMetrics {
+    pub fn new(sender: FutureScheduler<PdTask>) -> ExecLocalMetrics {
+        ExecLocalMetrics {
+            flow_stats: CopFlowStatistics::new(sender),
+            scan_details: Default::default(),
+            scan_counter: Default::default(),
+            exec_counter: Default::default(),
+        }
+    }
+
+    pub fn finish_task(
+        &mut self,
+        type_str: &'static str,
+        region_id: u64,
+        mut metrics: ExecutorMetrics,
+    ) {
+        let stats = &metrics.cf_stats;
+        // cf statistics group by type
+        self.scan_details.add(type_str, stats);
+        // flow statistics group by region
+        self.flow_stats.add(region_id, stats);
+        // scan count
+        self.scan_counter.merge(&mut metrics.scan_counter);
+        // executor count
+        self.exec_counter.merge(&mut metrics.executor_count);
+    }
+
+    pub fn flush(&mut self) {
+        self.flow_stats.flush();
+        self.scan_details.flush();
+        self.scan_counter.flush();
+        self.exec_counter.flush();
+    }
+}
+
+/// `ComLocalMetrics` is for the common metrics for coprocessor requests.
```

Comment:
em, seem we should use a better name instead of Com.

Reply:
Maybe `BasicLocalMetrics`? @siddontang